run_tests.py  --runs_per_test functionality is broken

Because of recently added sampling feature:

Trying to run without sampling with `-n 2`:
```
$ tools/run_tests/run_tests.py -l c -n 2
PASSED: make [time=11.5sec; retries=0:0]2017-03-08 20:25:16,989 detected port server running version 9
2017-03-08 20:25:17,019 my port server is version 9
Traceback (most recent call last):
  File "tools/run_tests/run_tests.py", line 1536, in <module>
    build_only=args.build_only)
  File "tools/run_tests/run_tests.py", line 1471, in _build_and_run
    assert args.runs_per_test == 1, "Can't do sampling (-p) over multiple runs (-n)."
AssertionError: Can't do sampling (-p) over multiple runs (-n).
```

see https://github.com/grpc/grpc/pull/9741

--- a/tools/run_tests/run_tests.py
+++ b/tools/run_tests/run_tests.py
@@ -1465,10 +1465,9 @@ def _build_and_run(
       sample_size = int(num_jobs * args.sample_percent/100.0)
       massaged_one_run = random.sample(massaged_one_run, sample_size)
       if not isclose(args.sample_percent, 100.0):
+        assert args.runs_per_test == 1, "Can't do sampling (-p) over multiple runs (-n)."
         print("Running %d tests out of %d (~%d%%)" %
               (sample_size, num_jobs, args.sample_percent))
-      else:
-        assert args.runs_per_test == 1, "Can't do sampling (-p) over multiple runs (-n)."
     if infinite_runs:
       assert len(massaged_one_run) > 0, 'Must have at least one test for a -n inf run'
     runs_sequence = (itertools.repeat(massaged_one_run) if infinite_runs

